### PR TITLE
Add Dotnet support to build-info-go

### PR DESCRIPTION
**Pull request by gailazar300**
- [x] All [tests](https://github.com/jfrog/build-info-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/build-info-go/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I'm only starting to review the code, but I already noticed that the README should be updated.

File: cli/cli.go [hunk 1]
```diff
@@ -146,6 +146,64 @@ func GetCommands(logger utils.Log) []*clitool.Command {
...
+				}
+				err = nugetModule.Build()
+				if err != nil {
+					return
+				}
+				return printBuild(bld, context.String(formatFlag))
+			},
+		},
+		{
+			Name:      "dotnet",
```
Comment (eyalbe4) [inline cli/cli.go:179]: I suggest removing the `dotnet` CLI command, laving only `nuget`.

File: utils/regexputils.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package utils
+
```
Comment (eyalbe4) [inline utils/regexputils.go:2]: Let's rename this file to utils.go.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
It looks like the `RunGo` function on `goutils.go` is unused. Let's try removing it.

File: utils/fileutils.go [hunk 3]
```diff
@@ -181,6 +182,50 @@ func ListFiles(path string, includeDirs bool) ([]string, error) {
 	return fileList, nil
 }
 
+// Return all files in the specified path who satisfy the filter func. Not recursive.
+func ListFilesByFilterFunc(path string, filterFunc func(filePath string) (bool, error)) ([]string, error) {
```
[the remark this answers is not in the document]
Comment (gailazar300) [reply utils/fileutils.go:186]: The function 'ListFiles' was already there(in build-info-go), and from what I spoke with Asaf there was a big discussion about this issue and you agreed that it belongs there.
We already moved 'walk' to gofrog, so maybe it makes sense to move them both.

**Pull request merged**